### PR TITLE
Update psr/log

### PR DESCRIPTION
**Pull request by snapshotpl**

### CODE REVIEW

**Review by adamkawala — APPROVED** [review]

**Review by crash2k — APPROVED** [review]

**Comment by snapshotpl** [comment]
@maciejlew when we can expect review, merge and release?

**Comment by ProteanCode** [comment]
We need this in newer Symfony-based frameworks

File: composer.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "require": {
         "php": "^7 || ^8.0",
         "ext-json": "*",
-        "psr/log": "^1",
+        "psr/log": "^1 || ^2 || ^3",
```
Comment (maciejlew) [inline composer.json:21]: Run `make test-php8` and you will see:
```
Fatal error: Declaration of Smsapi\Client\Curl\SmsapiHttpClient::setLogger(Psr\Log\LoggerInterface $logger) must be compatible with Psr\Log\LoggerAwareInterface::setLogger(Psr\Log\LoggerInterface $logger): void in /app/src/Curl/SmsapiHttpClient.php on line 33
```
To fix that you have to drop PHP 7.0 support. Do that and I will release next major version with your changes.

**Review by maciejlew — CHANGES_REQUESTED** [review]

**Comment by maciejlew** [comment]
I'm not sure what action to be taken you expect from me. GitHub Actions are enabled as described here: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#about-github-actions-permissions-for-your-repository, however, I don't see any approve workflow run request as described here: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks. Is that what you expect? Maybe that approve workflow run do not exist because there is no workflow yet?

**Comment by snapshotpl** [comment]
Can you check if GH Actions are enabled here https://github.com/smsapi/smsapi-php-client/settings/actions ?

**Comment by snapshotpl** [comment]
Oh, see now you have already check this. Strange, workflow should works now

**Comment by snapshotpl** [comment]
Just added sample workflow from https://docs.github.com/en/actions/quickstart and still nothing. Probably something still need to be setting up.

**Comment by snapshotpl** [comment]
Now working and all test pass!

**Comment by maciejlew** [comment]
Glad to see that. I'm not sure what has helped, I haven't changed any setting. Maybe one of my PRs with workflow triggered something.

Anyway, despite that unit tests are passing, there is still problem with integration ones. HTTP client is being produced and error I've mentioned before occur only when running integration tests. Check out SmsapiClientIntegrationTestCase::prepare.

To run integration test you have to pass access token and that is the reason why we do not execute them here.

You can configure that tests locally, in your developement environment, even passing incorrect access token, and you will see the error.

**Comment by snapshotpl** [comment]
Check now.

I think we should run integration test on CI, you can provide token by secret, so will be not shared.

**Comment by snapshotpl** [comment]
LoggerAware is fixed, but founded custom test logger now.

**Comment by snapshotpl** [comment]
I think we should avoid current TestLogger - it's just echo debug information so looks like only for development proces. The same can be done by xdebug.

If we want to break version, then we need to release client 4 with log v2 and client 5 with log v3. So I prefer to drop TestLogger.

**Comment by maciejlew** [comment]
Running integration tests seems not to be a good decision for test stability and security reasons. 

Test may fail and test cleanup may not execute. That will make impact on future test runs.

Someone may write malicious test code that, for example, may send spam using our test account. Test are now being executed on each pull request so code review will always happen after the fact. Even not accepted pull request may cause security breach.

**Comment by maciejlew** [comment]
Drop TestLogger.

**Review by maciejlew — APPROVED** [review]

**Pull request merged**